app/helpers/application_helper/button/middleware_server_action.rb

```diff
@@ -1,7 +1,20 @@
 class ApplicationHelper::Button::MiddlewareServerAction < ApplicationHelper::Button::Basic
 
   def visible?
-    !@record.present? ||
-      (@record.try(:product) != 'Hawkular' && @record.try(:middleware_server).try(:product) != 'Hawkular')
+    if @record.present?
+      hawkular_product? ? false : !immutable?
```

Comment:
Can this part be simplified? 
perhaps something like "return record.present? && hawkular_product? && immutable?"